src/github.com/matrix-org/dendrite/roomserver/state/state.go

```diff
@@ -182,6 +195,79 @@ func DifferenceBetweeenStateSnapshots(db RoomStateDatabase, oldStateNID, newStat
 	}
 }
 
+// StringTuplesToNumericTuples converts the string state key tuples into numeric IDs
+// If there isn't a numeric ID for either the event type or the event state key then the tuple is discarded.
```

Comment:
I've combined StringTuplesToNumericTuples and LoadStateAtSnapshotForNumericTuples into a single function. This should make it clearer why we discard unknown event-types and state keys.